packages/addons-inject/tests/wrangler.test.js

```diff
@@ -0,0 +1,256 @@
+import { fetchMock, SELF } from "cloudflare:test";
+import { afterEach, beforeAll, describe, expect, it } from "vitest";
+
+import { AddonsConstants } from "../index";
+
+beforeAll(() => {
+  // Enable outbound request mocking...
+  fetchMock.activate();
+  // ...and throw errors if an outbound request isn't mocked
+  fetchMock.disableNetConnect();
+});
+
+// Ensure we matched every mock we defined
+afterEach(() => fetchMock.assertNoPendingInterceptors());
+
+describe("Addons when disabled", async () => {
+  it("skips injection on HTML based on headers", async () => {
+    fetchMock
+      .get("http://test-builds.devthedocs.org")
+      .intercept({ path: "/en/latest/" })
+      .reply(200, "<html></html>", {
+        headers: {
+          "Content-type": "text/html",
+          "X-RTD-Force-Addons": false,
+          "X-RTD-Hosting-Integrations": false,
+        },
+      });
+    let response = await SELF.fetch(
+      "http://test-builds.devthedocs.org/en/latest/",
+    );
+    expect(response.status).toBe(200);
+    expect(await response.text()).toBe("<html></html>");
+  });
+});
+
+describe("Addons when enabled", async () => {
+  it("handles HTML injection and no rewrites", async () => {
+    fetchMock
+      .get("http://test-builds.devthedocs.org")
+      .intercept({ path: "/en/latest/" })
+      .reply(200, "<html><head></head><body></body></html>", {
+        headers: {
+          "Content-type": "text/html",
+          "X-RTD-Force-Addons": true,
+          "X-RTD-Hosting-Integrations": false,
+        },
+      });
+    let response = await SELF.fetch(
+      "http://test-builds.devthedocs.org/en/latest/",
+    );
+    expect(response.status).toBe(200);
+    expect(await response.text()).toBe(
+      `<html><head>${AddonsConstants.scriptAddons}</head><body></body></html>`,
+    );
```

Comment:
We can add checks here for the headers `X-RTD-Force-Addons` and `X-RTD-Hosting-Integrations` as well; so we know they are passing through as basic checks.